sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -53,17 +54,18 @@
 @ServiceClient(builder = DigitalTwinsClientBuilder.class, isAsync = true)
 public final class DigitalTwinsAsyncClient {
     private static final ClientLogger logger = new ClientLogger(DigitalTwinsAsyncClient.class);
-    private static final ObjectMapper mapper = new ObjectMapper();
+    private ObjectMapper mapper;
     private final DigitalTwinsServiceVersion serviceVersion;
     private final AzureDigitalTwinsAPIImpl protocolLayer;
     private static final Boolean includeModelDefinitionOnGet = true;
     private final JsonSerializer serializer;
 
     DigitalTwinsAsyncClient(String serviceEndpoint, HttpPipeline pipeline, DigitalTwinsServiceVersion serviceVersion, JsonSerializer jsonSerializer) {
         final SimpleModule stringModule = new SimpleModule("String Serializer");
-        stringModule.addSerializer(new DigitalTwinsStringSerializer(String.class, mapper));
 
         JacksonAdapter jacksonAdapter = new JacksonAdapter();
+        mapper = jacksonAdapter.serializer(); // Use the same mapper in this layer that the generated layer will use
+        stringModule.addSerializer(new DigitalTwinsStringSerializer(String.class, mapper));
```

Comment:
Do we still need this string serializer? Are there any APIs that still operate on Strings? I thought we got rid of all those, and everything was objects now.

Reply:
That was the plan back when we were going to force APIs to take in a <T extends IDigitalTwin>, but those plans fell through, so we're back to supporting strings again. It would be more work to remove them at this point then I think it would be worth

Reply:
Got it, so now we are back to the original state where we support `<T>` as `String` as input for these APIs.